File: node/lib/lex-result.ts
import { Result } from './result.js'
import { TokenList } from './token-list.js'

export class LexResult extends Result {
  readonly value: TokenList

  constructor(value: any[], source: string, warnings: any[] = [], errors: any[] = []) {
    super(source, warnings, errors)
    this.value = new TokenList(value)
  }

  override success(): boolean {
    return this.errors.length === 0
  }

  override failed(): boolean {
    return this.errors.length > 0
  }
}

File: lib/herb/lex_result.rb
# frozen_string_literal: true
# typed: true

module Herb
  class LexResult < Result
    attr_reader :value #: TokenList

    #: (Array[Herb::Token], String, Array[Herb::Warnings::Warning], Array[Herb::Errors::Error]) -> void
    def initialize(value, source, warnings, errors)
      @value = TokenList.new(value)
      super(source, warnings, errors)
    end
  end
end
